### DISCUSSION TITLE
Auto detect of realm

### DISCUSSION THREAD

**Discussion by JackisGT** [PierreGode/Linux-Active-Directory-join-script#79] [Q&A] [answered]
how does the script auto detect a realm?

**Comment by PierreGode** [comment, marked as answer]
The script auto-detects a realm by using the realm discover command, which queries the network for available domains. Here’s how it works:

Network Query: realm discover sends a request to locate domains that are configured for integration with Linux systems.

Domain Information: If a domain is properly configured for discovery, realm discover will retrieve details like the domain name, associated domain controllers, and any supported authentication mechanisms (e.g., Kerberos).

Auto-Detection: Based on the response from the network, the script identifies if the specified domain is reachable and properly set up for Active Directory integration.

This process enables the script to dynamically check for accessible domains without manual input, ensuring that it only proceeds if the domain is available and ready for a connection
